pkg/program/shell/run_dir_unix_test.go

```diff
@@ -0,0 +1,102 @@
+// +build !windows,!plan9
+
+package shell
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/elves/elvish/pkg/util"
+)
+
+// RunDirPathTest is a test case to be used in getRunDirPath.
+type RunDirPathTest struct {
+	Name          string
+	XDGRuntimeDir string
+	Tmpdir        string
+	Want          string
+}
+
+func TestGetSecureRunDir_UseXDGWhenPresent(t *testing.T) {
```

Comment:
You do not need to test `getRunDirPath`. You can test `getSecureRunDir` if you do the following:

* Create **two** temporary directories and use them as overrides for **both** of `XDG_RUNTIME_DIR` and `TMPDIR`.

* Since the expected run dir now depends on the path of the temporary directories, move the test table inside the function, after the creation of the temporary directories.

There are 4 cases to test:

* XDG_RUNTIME_DIR set, TMPDIR set
* XDG_RUNTIME_DIR set, TMPDIR unset
* XDG_RUNTIME_DIR unset, TMPDIR set
* XDG_RUNTIME_DIR unset, TMPDIR unset

However, the last case will use the system temporary directory `/tmp`, which will likely already have an existing Elvish run dir, so you can skip that case.

The unit test should also check if the run dir has the correct permission.